waspc/cli/src/Wasp/Cli/Interactive.hs

```diff
@@ -0,0 +1,86 @@
+module Wasp.Cli.Interactive
+  ( askForInput,
+    askToChoose,
+    askForRequiredInput,
+  )
+where
+
+import Data.Foldable (find)
+import Data.List (intercalate)
+import qualified Data.Text as T
+import System.IO (hFlush, stdout)
+import Text.Read (readMaybe)
+import qualified Wasp.Util.Terminal as Term
+
+askForRequiredInput :: String -> IO String
+askForRequiredInput question = do
+  ensureNotNull . askForInput $ question
+
+askToChoose :: Show a => String -> [a] -> IO a
+askToChoose question options = case options of
+  [] -> error "Cannot ask to choose from empty list of options."
+  [singleOption] -> return singleOption
+  _multipleOptionsCase -> askUserToChoose
+  where
+    askUserToChoose = do
+      putStrLn $ Term.applyStyles [Term.Bold] question
+      putStrLn optionsToChooseFrom
+      answer <- prompt
+
+      handleOptionIndex answer >>= \case
+        Just option -> return option
+        Nothing -> handleOptionName answer
+
+    handleOptionIndex answer = do
```

Comment:
handleOptionIndex is not the best name -> how does it handle it, what does it do?